kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/domain.kt

```diff
@@ -0,0 +1,164 @@
+package io.kotest.property.arbitrary
+
+import io.kotest.property.Arb
+
+/**
+ * Generates random domains
+ *
+ * https://en.wikipedia.org/wiki/Domain_name#Domain_name_syntax
+ * https://whogohost.com/host/knowledgebase/308/Valid-Domain-Name-Characters.html
+ */
+fun Arb.Companion.domain(
+   tlds: List<String> = top120TLDs,
+   labelArb: Arb<String> = _labelArb
+): Arb<String> = arbitrary { rs ->
+   val amountOfLabels = rs.random.nextInt(1, 4)
+
+   val labels = labelArb.take(amountOfLabels).joinToString(separator = ".", postfix = ".")
```

Comment:
Do we want domains to end with a dot ?

Reply:
They don't. 
It's just the label (parts before the TLD). So we take 1 to 3 labels (a.b.c) and then add a dot and a tld (a.b.c.com)